druntime/src/core/math.d

```diff
@@ -267,3 +269,1298 @@ T toPrec(T:real)(real f)  { pragma(inline, false); return f; }
+                [0x1.9200000000000p-1, 0, 0x1.ef5ade4dd0000p-3, -0x1.a211565bb8e11p-51],
+                [0x1.9000000000000p-1, 0, 0x1.f991c6cb3b000p-3, 0x1.bcbecca0cdf30p-46],
+                [0x1.8c00000000000p-1, 0, 0x1.07138604d5800p-2, 0x1.89cdb16ed4e91p-48],
+                [0x1.8a00000000000p-1, 0, 0x1.0c42d67616000p-2, 0x1.7188b163ceae9p-45],
+                [0x1.8800000000000p-1, 0, 0x1.1178e8227e800p-2, -0x1.c210e63a5f01cp-45],
+                [0x1.8600000000000p-1, 0, 0x1.16b5ccbacf800p-2, 0x1.b9acdf7a51681p-45],
+                [0x1.8400000000000p-1, 0, 0x1.1bf99635a6800p-2, 0x1.ca6ed5147bdb7p-45],
+                [0x1.8200000000000p-1, 0, 0x1.214456d0eb800p-2, 0x1.a87deba46baeap-47],
+                [0x1.7e00000000000p-1, 0, 0x1.2bef07cdc9000p-2, 0x1.a9cfa4a5004f4p-45],
+                [0x1.7c00000000000p-1, 0, 0x1.314f1e1d36000p-2, -0x1.8e27ad3213cb8p-45],
+                [0x1.7a00000000000p-1, 0, 0x1.36b6776be1000p-2, 0x1.16ecdb0f177c8p-46],
+                [0x1.7800000000000p-1, 0, 0x1.3c25277333000p-2, 0x1.83b54b606bd5cp-46],
+                [0x1.7600000000000p-1, 0, 0x1.419b423d5e800p-2, 0x1.8e436ec90e09dp-47],
+                [0x1.7400000000000p-1, 0, 0x1.4718dc271c800p-2, -0x1.f27ce0967d675p-45],
+                [0x1.7200000000000p-1, 0, 0x1.4c9e09e173000p-2, -0x1.e20891b0ad8a4p-45],
+                [0x1.7000000000000p-1, 0, 0x1.522ae0738a000p-2, 0x1.ebe708164c759p-45],
+                [0x1.6e00000000000p-1, 0, 0x1.57bf753c8d000p-2, 0x1.fadedee5d40efp-46],
+                [0x1.6c00000000000p-1, 0, 0x1.5d5bddf596000p-2, -0x1.a0b2a08a465dcp-47],
+            ];
+
+            // x = 2^^k z; where z is in range [OFF,2*OFF) and exact.
+            // The range is split into N subintervals.
+            // The ith subinterval contains z and c is near its center.
+            immutable tmp = ix - 0x3FE6955500000000UL;
+            immutable i = (tmp >> 45) % log_tab.length;
+            immutable iz = ix - (tmp & 0xFFFUL << 52);
+            immutable dz = *cast(double*)&iz;
+            immutable k = cast(double)(cast(long)tmp >> 52);
+            // log(x) = k*Ln2 + log(c) + log1p(z/c-1).
+            immutable invc = log_tab[i][0];
+            immutable logc = log_tab[i][2];
+            immutable logctail = log_tab[i][3];
+            // Split z such that rhi, rlo and rhi*rhi are exact and |rlo| <= |r|.
+            immutable izhi = (iz + (1UL << 31)) & (-1UL << 32);
+            immutable zhi = *cast(double*)&izhi;
+            immutable zlo = dz - zhi;
+            immutable rhi = zhi * invc - 1.0;
+            immutable rlo = zlo * invc;
+            immutable r = rhi + rlo;
+            // k*Ln2 + log(c) + r.
+            immutable t1 = k * ln2hi + logc;
+            immutable t2 = t1 + r;
+            immutable lo1 = k * ln2lo + logctail;
+            immutable lo2 = t1 - t2 + r;
+            // Evaluation is optimized assuming superscalar pipelined execution.
```

Comment:
multiple issue? All processors are pipelined in some way and basically anything is superscalar now.